cmd/settlement_paypal.go

```diff
@@ -0,0 +1,323 @@
+	must(viper.BindEnv("rate-auth", "RATE_AUTH"))
+	must(transformPaypalSettlementCmd.MarkPersistentFlagRequired("rate-auth"))
+
+	// rate
+	paypalSettlementCmd.PersistentFlags().Float64VarP(&rate, "rate", "r", 0,
+		"the rate to compute the currency conversion")
+	must(viper.BindPFlag("rate", paypalSettlementCmd.PersistentFlags().Lookup("rate")))
+	must(viper.BindEnv("rate", "RATE"))
+	must(transformPaypalSettlementCmd.MarkPersistentFlagRequired("rate"))
+}
+
+// EmailTemplate performs template replacement of date fields in emails
+func EmailTemplate(inPath string, outPath string) (err error) {
+	// read in email template
+	data, err := ioutil.ReadFile(inPath)
+	if err != nil {
+		err = fmt.Errorf("failed to read template: %w", err)
+		return
+	}
+	// perform template rendering to out
+	f, err := os.Create(outPath)
+	if err != nil {
+		err = fmt.Errorf("failed to create output: %w", err)
+		return
+	}
+	defer func() {
+		if err = f.Close(); err != nil {
+			err = fmt.Errorf("failed to create output: %w", err)
+			return
+		}
+	}()
+
+	var (
+		today = time.Now()
+		// template will have a "year" and "month" field
+		v = struct {
+			Month int
+			Year  int
+		}{
+			Month: int(today.Month()),
+			Year:  today.Year(),
+		}
+		t = template.Must(template.New("email").Parse(string(data)))
+	)
+
+	if err = t.Execute(f, v); err != nil {
+		err = fmt.Errorf("failed to execute template: %w", err)
+		return
+	}
+	return
+}
+
+var (
+	paypalSettlementCmd = &cobra.Command{
+		Use:   "paypal",
+		Short: "provides paypal settlement",
+	}
+
+	emailPaypalSettlementCmd = &cobra.Command{
+		Use:   "email",
+		Short: "provides population of a templated email",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := EmailTemplate(input, out); err != nil {
+				log.Printf("failed to perform email templating: %s\n", err)
+				os.Exit(1)
+			}
+		},
+	}
+
+	completePaypalSettlementCmd = &cobra.Command{
+		Use:   "complete",
+		Short: "provides completion of paypal settlement",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := CompleteSettlement(input, out, txnID); err != nil {
+				log.Printf("failed to perform complete: %s\n", err)
+				os.Exit(1)
+			}
+		},
+	}
+
+	transformPaypalSettlementCmd = &cobra.Command{
+		Use:   "transform",
+		Short: "provides transform of paypal settlement for mass pay",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := TransformForMassPay(TransformArgs{
+				In:       input,
+				Currency: currency,
+				Auth:     auth,
+				Rate:     decimal.NewFromFloat(rate),
+				Out:      out,
+			}); err != nil {
+				log.Printf("failed to perform transform: %s\n", err)
+				os.Exit(1)
+			}
+		},
+	}
+)
+
+// CompleteSettlement marks the settlement file as complete
+func CompleteSettlement(inPath string, outPath string, txnID string) error {
+	fmt.Println("RUNNING: complete")
+	if inPath == "" {
+		return errors.New("the 'in' flag must be set")
```

Comment:
I think this flag is now `-i` shorthand?

Reply:
seems like txnID mentions might be similarly out of date

Reply:
resolved in https://github.com/brave-intl/bat-go/pull/337/commits/e612c3fd66e8f2a2e905979c2133adaa0ff3519f